php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -818,15 +828,18 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         } else {
             $oldData = [];
         }
+
         $newData = array_merge($oldData ?? [], $values);
 
+        $flagDataToUpdate['Data'] = empty($newData) ? null : json_encode($newData);
+
         // Save the JSON to the flag.Data column.
         //
         // json_encode ensures that this is safe. If we use the safe wrapper,
         // HTML encoding the quotation marks will make it invalid JSON.
         $db->unsafeUpdate(
```

Comment:
@CamilleBeau there is no reason to update the `data_entry_completion_status` in an unsafe way. leave that for the instrument, make a different query for this field

Reply:
@ridz1208 Figured out why unsafeUpdate was being used

>  // json_encode ensures that this is safe. If we use the safe wrapper,
>  // HTML encoding the quotation marks will make it invalid JSON.

HTML double escape for the Data column in flag